src/osemgrep/networking/Semgrep_App.ml

```diff
@@ -215,21 +223,25 @@ let start_scan ~dry_run caps (prj_meta : Project_metadata.t)
       body |> Yojson.Basic.from_string |> Yojson.Basic.pretty_to_string
     in
     Logs.debug (fun m -> m "Starting scan: %s" pretty_body);
-    match Http_helpers.post ~body ~headers caps#network url with
-    | Ok body -> extract_scan_id body
-    | Error (status, msg) ->
+    match%lwt Http_helpers.post ~body ~headers caps#network url with
+    | Ok { body = Ok body; _ } -> Lwt.return (extract_scan_id body)
+    | Ok { body = Error err; code; _ } ->
         let pre_msg =
-          if status =|= 404 then
+          if code =|= 404 then
             {|Failed to create a scan with given token and deployment_id.
 Please make sure they have been set correctly.
 |}
           else ""
         in
         let msg =
           Fmt.str "%sAPI server at %a returned this error: %s" pre_msg Uri.pp
```

Comment:
Seems like we should pick one of `spf` or `Fmt.str` to be consistent with.

Reply:
I'll switch to spf